datumaro/plugins/transforms.py

```diff
@@ -564,6 +567,66 @@ def transform_item(self, item):
                 annotations.append(ann.wrap())
         return item.wrap(annotations=annotations)
 
+class ProjectLabels(ItemTransform):
+    """
+    Changes the order of labels in the dataset from the existing
+    to the desired one and removes unknown labels. Updates or removes
+    the corresponding annotations.
+
+    Labels are matched by names (case dependent).
+
+    Useful for merging similar datasets.
+    """
+
+    def __init__(self, extractor: IExtractor, dst_labels: LabelCategories):
+        super().__init__(extractor)
+
+        self._categories = {}
+
+        src_label_cat = self._extractor.categories().get(AnnotationType.label)
+        self._make_label_id_map(src_label_cat, dst_labels)
+
+        src_mask_cat = self._extractor.categories().get(AnnotationType.mask)
+        if src_mask_cat is not None:
+            assert src_label_cat is not None
+            dst_mask_cat = MaskCategories(attributes=src_mask_cat.attributes)
+            dst_mask_cat.colormap = {
+                id: src_mask_cat[id]
+                for id, _ in enumerate(src_label_cat.items)
+                if id in src_mask_cat and (self._map_id(id) or id == 0)
+            }
+            self._categories[AnnotationType.mask] = dst_mask_cat
+
+        src_point_cat = self._extractor.categories().get(AnnotationType.points)
+        if src_point_cat is not None:
+            assert src_label_cat is not None
+            dst_point_cat = PointsCategories(attributes=src_point_cat.attributes)
+            dst_point_cat.items = {
+                id: src_point_cat[id]
+                for id, _ in enumerate(src_label_cat.items)
+                if id in src_point_cat and (self._map_id(id) or id == 0)
+            }
+            self._categories[AnnotationType.points] = dst_point_cat
+
+    def _make_label_id_map(self, src_label_cat, dst_label_cat):
+        self._map_id = lambda src_id: \
+            dst_label_cat.find(src_label_cat[src_id].name)[0]
+        self._categories[AnnotationType.label] = dst_label_cat
+
+    def categories(self):
+        return self._categories
+
+    def transform_item(self, item):
+        annotations = []
+        for ann in item.annotations:
+            if getattr(ann, 'label') is not None:
```

Comment:
`getattr(ann, 'label')` is the same as `ann.label`. You probably meant to specify a `None` default value.